From 0a536a93a414cfe1509283df8e133f3edf732cb6 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Fri, 18 Feb 2022 12:51:15 -0700
Subject: [PATCH] Revert "BACKPORT: FROMLIST: Makefile: lld: set -O2 linker
 flag when linking with LLD"

This reverts commit 4a4ea24df22cf8419006f5f65745f12b5fcdace0.

Link: https://github.com/ClangBuiltLinux/linux/issues/1578
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 Makefile | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/Makefile b/Makefile
index 45d7fc5c47d2..d6faa32d03b4 100644
--- a/Makefile
+++ b/Makefile
@@ -751,10 +751,6 @@ endif
 # Use make W=1 to enable them (see scripts/Makefile.extrawarn)
 KBUILD_CFLAGS += $(call cc-disable-warning, unused-but-set-variable)
 
-ifeq ($(ld-name),lld)
-LDFLAGS += -O2
-endif
-
 KBUILD_CFLAGS += $(call cc-disable-warning, unused-const-variable)
 ifdef CONFIG_FRAME_POINTER
 KBUILD_CFLAGS	+= -fno-omit-frame-pointer -fno-optimize-sibling-calls
-- 
2.35.1

